### ISSUE TITLE
Freestanding platform

### ISSUE THREAD

**Issue by RossComputerGuy** [dart-lang/sdk#56958]
It would be great if Dart could be compiled in a "freestanding" way, this means compile without any OS assumptions. It could make it easier to embed Dart into unique places like UEFI without patching the compiler. The embedder would then be responsible for providing the platform API's. Updating the GN stuff to provide static libraries of the Dart API and runtime for embedding would be great as well.

**Comment by a-siva** [comment]
@RossComputerGuy thanks for the suggestion, at this point we don't have the cycles to undertake a project like this. We will keep for future reference.

**Comment by mraleph** [comment]
> It would be great if Dart could be compiled in a "freestanding" way, this means compile without any OS assumptions.

Runtime needs access to some very specific OS features (like virtual memory management, threading, signal handling etc). Compiler needs to be aware about platform specific ABIs. Abstracting all this away into a neat embedding API is a rather hard. I don't think this will ever happen to be honest. 

> Updating the GN stuff to provide static libraries of the Dart API and runtime for embedding would be great as well.

This work is actually in progress by @iinozemtsev, we would like to provide tooling and libraries in the Dart SDK to enable this.

**Comment by RossComputerGuy** [comment]
> Abstracting all this away into a neat embedding API is a rather hard. I don't think this will ever happen to be honest.

True, maybe just having a minimal runtime would work? Just enough to execute code.
